server/datastore/mysql/statistics.go

```diff
@@ -55,44 +73,40 @@ func (ds *Datastore) ShouldSendStatistics(ctx context.Context, frequency time.Du
 			if err != nil {
 				return fleet.StatisticsPayload{}, false, ctxerr.Wrap(ctx, err, "insert statistics")
 			}
-			return fleet.StatisticsPayload{
-				AnonymousIdentifier:       anonIdentifier,
-				FleetVersion:              version.Version().Version,
-				LicenseTier:               license.Tier,
-				NumHostsEnrolled:          amountEnrolledHosts,
-				NumUsers:                  amountUsers,
-				NumTeams:                  amountTeams,
-				NumPolicies:               amountPolicies,
-				NumLabels:                 amountLabels,
-				SoftwareInventoryEnabled:  appConfig.HostSettings.EnableSoftwareInventory,
-				VulnDetectionEnabled:      appConfig.VulnerabilitySettings.DatabasesPath != "",
-				SystemUsersEnabled:        appConfig.HostSettings.EnableHostUsers,
-				HostsStatusWebHookEnabled: appConfig.WebhookSettings.HostStatusWebhook.Enable,
-			}, true, nil
+
+			// compute active weekly users since now - frequency
+			stats := fleet.StatisticsPayload{
+				AnonymousIdentifier: anonIdentifier,
+				FleetVersion:        version.Version().Version,
+				LicenseTier:         license.Tier,
+			}
+			if err := computeStats(&stats, time.Now().Add(-frequency)); err != nil {
```

Comment:
Would it be more precise for `since` to be the time that the statistics were recorded as last sent rather than now - frequency (in case there was a delay in successfully posting the stats)?

Reply:
You're right, but that's how it's done - it's just that on this line, this is the case where the statistics were never sent (loading the stats metadata returned `sql.ErrNoRows`). The case you mention is below, and uses `lastUpdated` as `since` time.

Reply:
Ah, I see now 🤦🏻‍♀️